mig-agent/context.go

```diff
@@ -285,6 +286,19 @@ func initAgentID(orig_ctx Context) (ctx Context, err error) {
 			panic(err)
 		}
 	}
+	// Make sure the obtained queue location matches the format that we expect, if
+	// it doesn't create a new one
+	mtch, err := regexp.Match("^\\S+\\.\\S+$", id)
```

Comment:
This is going to break backward compat and all agents will create a new queueloc.
I _think_ this won't have too much impact, but it's friday and I'm drinking a 10% beer...